hudi-common/src/main/java/org/apache/hudi/common/util/ParquetUtils.java

```diff
@@ -283,17 +283,37 @@ public Boolean apply(String recordKey) {
 
   /**
    * Parse min/max statistics stored in parquet footers for all columns.
+   * ParquetRead.readFooter is not a thread safe method.
+   *
+   * @param conf hadoop conf.
+   * @param parquetFilePath file to be read.
+   * @param cols cols which need to collect statistics.
+   * @param useLock if use lock when read parquet footer.
+   * @return a HoodieColumnRangeMetadata instance.
    */
-  public Collection<HoodieColumnRangeMetadata<Comparable>> readRangeFromParquetMetadata(Configuration conf, Path parquetFilePath, List<String> cols) {
-    ParquetMetadata metadata = readMetadata(conf, parquetFilePath);
+  public Collection<HoodieColumnRangeMetadata<Comparable>> readRangeFromParquetMetadata(
+      Configuration conf,
+      Path parquetFilePath,
+      List<String> cols,
+      boolean useLock) {
```

Comment:
I am so curious about the change, would you please point me to the docs that read footer is not thread-safe and affect the result?

Reply:
We get the statistics of the specified column by reading footer.

However, if the specified column is of datetye type, parquet uses simpledateformat to format data（this is not thread safe） when returning the min-max value. In the case of multithreading, this will lead to the return of the wrong min max value.

Let me add a test case

Reply:
Reduce the scope of locking by using a private static lock.   only when we counter DateType, we locked